test/functional/wallet_sendall.py

```diff
@@ -394,11 +394,48 @@ def sendall_spends_unconfirmed_inputs_if_specified(self):
         self.log.info("Test that sendall spends specified unconfirmed inputs")
         self.def_wallet.sendtoaddress(self.wallet.getnewaddress(), 17)
         assert_greater_than(self.wallet.getbalances()["mine"]["untrusted_pending"], 0)
+        self.wallet.syncwithvalidationinterfacequeue()
         unspent = self.wallet.listunspent(minconf=0)[0]
 
         self.wallet.sendall(recipients=[self.remainder_target], inputs=[unspent])
         assert_equal(self.wallet.getbalance(), 0)
 
+    @cleanup
+    def sendall_does_ancestor_aware_funding(self):
+        self.log.info("Test that sendall does ancestor aware funding for unconfirmed inputs")
+
+        # higher parent feerate
+        self.def_wallet.sendtoaddress(address=self.wallet.getnewaddress(), amount=17, fee_rate=20)
+        self.wallet.syncwithvalidationinterfacequeue()
+
+        assert_greater_than(self.wallet.getbalances()["mine"]["untrusted_pending"], 0)
+        unspent = self.wallet.listunspent(minconf=0)[0]
+
+        parent_txid = unspent["txid"]
+        assert_equal(self.wallet.gettransaction(parent_txid)["confirmations"], 0)
+
+        res_1 = self.wallet.sendall(recipients=[self.def_wallet.getnewaddress()], inputs=[unspent], fee_rate=20, add_to_wallet=False, lock_unspents=True)
+        child_hex = res_1["hex"]
+
+        child_tx = self.wallet.decoderawtransaction(child_hex)
+        higher_parent_feerate_amount = child_tx["vout"][0]["value"]
+
+        # lower parent feerate
+        self.def_wallet.sendtoaddress(address=self.wallet.getnewaddress(), amount=17, fee_rate=10)
+        assert_greater_than(self.wallet.getbalances()["mine"]["untrusted_pending"], 0)
```

Comment:
Same as before, would be better to not use `assert_greater_than`, the wallet is receiving exactly 17 btc.

Reply:
Done, but it is 34 because the wallet also received 17 previously.